src/backend/api/meals.js

```diff
@@ -1,15 +1,121 @@
 const express = require("express");
+const { on } = require("../database");
 const router = express.Router();
 const knex = require("../database");
 
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    if ("maxPrice" in request.query) {
+      const maxPrice = parseFloat(request.query.maxPrice)
+      if (isNaN(maxPrice)) {
+        response.status(400).json({ error: "maxPrice must be an integer" })
+        return
+      }
+      const MaxPriceMeals = await knex('meals')
+        .where("price", "<=", maxPrice)
+      response.json(MaxPriceMeals);
+      return
+    }
+    else if ("availableReservations" in request.query) {
+      const availableReservations = request.query.availableReservations;
+      if (availableReservations == "true") {
+
+        const mealswithAvailableReservation = await knex("meals")
+          .select("meals.id", "meals.title", "meals.max_reservations ")
```

Comment:
not sure if this is correct....
```suggestion
          .select("id", "title", "max_reservations ")
```
maybe ?